### PR TITLE
Observer doc improvements

### PR DESCRIPTION
**Pull request by ashkan-saeedi-mazdeh**
Improved observer docs a bit, next step is to create a section for communication in the system which will contain streams, observers and grain-grain and client-grain communication and will have a page comparing observers and SMS streams

### CODE REVIEW

File: src/Documentation/Getting-Started-With-Orleans/Observers.md [hunk 1]
```diff
@@ -109,6 +109,9 @@ Now whenever our grain on the server calls the `SendUpdateMessage` method, all s
 
 **Note:** Objects passed to `CreateObjectReference` are held via a [`WeakReference<T>`](https://msdn.microsoft.com/en-us/library/system.weakreference) and will therefore be garbage collected if no other references exist. Users should maintain a reference for each observer which they do not want to be collected.
 
+**Note:** Observers are inherently unreliable since you don't get any response back to know if the message is received and processed or simply failed due to any condition which might arise in a distributed system. Because of that your observers should pull the grain periodically or use any other mechanism to ensure that they received all messages which they should have received.
```
Comment (ReubenBond) [inline src/Documentation/Getting-Started-With-Orleans/Observers.md]: s/pull/poll

File: src/Documentation/Getting-Started-With-Orleans/Observers.md [hunk 2]
```diff
@@ -109,6 +109,9 @@ Now whenever our grain on the server calls the `SendUpdateMessage` method, all s
...
+In some situations you can afford losing some messages and you don't need any additional mechanism but if you need to make sure that all observers are always receiving the messages and are receiving all of them, both periodic resubscriptions and pulling the observer grain, can help to ensure eventual processing of all messages.
```
Comment (ReubenBond) [inline src/Documentation/Getting-Started-With-Orleans/Observers.md]: s/losing/to lose
s/pulling/polling

**Comment by ashkan-saeedi-mazdeh** [comment]
@ReubenBond  Thanks and fixed.

**Comment by ReubenBond** [comment]
Big thanks, @ashkan-saeedi-mazdeh :)

**Pull request merged**